Modules/IO/GDCM/src/itkGDCMImageIO.cxx

```diff
@@ -331,6 +333,33 @@ GDCMImageIO::Read(void * pointer)
     // WARNING: sizeof(Real World Value) != sizeof(Stored Pixel)
     len = len * outputpt.GetPixelSize() / pixeltype.GetPixelSize();
   }
+  /*
+   * YBR to RGB
+   * Currently (GDCM 3.0.2) processing for lossy Jpeg YBR_FULL_422
+   * is not fully clear. May be later the files will be converted
+   * internally in GDCM, in that case remove YBR_FULL_422 below.
+   *
+   * TODO: UINT16
+   *
+   */
+  if (m_NumberOfComponents == 3 &&
+      (pi == gdcm::PhotometricInterpretation::YBR_FULL || pi == gdcm::PhotometricInterpretation::YBR_FULL_422) &&
+      m_YBRtoRGB && (pixeltype == gdcm::PixelFormat::UINT8 || pixeltype == gdcm::PixelFormat::INT8) && (len % 3 == 0))
+  {
+    unsigned char * copy = reinterpret_cast<unsigned char *>(pointer);
+    for (size_t x = 0; x < len; x += 3)
+    {
+      const double Y = static_cast<double>(copy[x]);
+      const double Cb = static_cast<double>(copy[x + 1]);
+      const double Cr = static_cast<double>(copy[x + 2]);
+      const int    R = static_cast<int>((Y + 1.402 * (Cr - 128)) + 0.5);
+      const int    G = static_cast<int>((Y - (0.114 * 1.772 * (Cb - 128) + 0.299 * 1.402 * (Cr - 128)) / 0.587) + 0.5);
+      const int    B = static_cast<int>((Y + 1.772 * (Cb - 128)) + 0.5);
+      copy[x] = static_cast<unsigned char>((R < 0) ? 0 : R);
```

Comment:
In GDCM I am doing a full clamp here (<= 255):

* https://github.com/malaterre/GDCM/blob/master/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h#L66

Reply:
I think it is not required with proper cast, it is many years tested code.

Reply:
@malaterre  If you  [comment two lines here in GDCM,](https://github.com/InsightSoftwareConsortium/ITK/blob/3526b67168ded206a8b9a3e7100fb1849c476d8e/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmJPEGBITSCodec.hxx#L858) convert wiil be not required for YBR_FULL_422, i have tried PR in GDCM and closed, please see yourself, may be it will have impact on some GDCM tools or whatever

Reply:
@malaterre 
```
      if( GetPhotometricInterpretation() == PhotometricInterpretation::YBR_FULL
      || GetPhotometricInterpretation() == PhotometricInterpretation::YBR_FULL_422 )
        {
// <------       cinfo.jpeg_color_space = JCS_UNKNOWN;
// <------        cinfo.out_color_space = JCS_UNKNOWN;
        //this->PlanarConfiguration = 1;
        }
```